### PR TITLE
renpy: init at 8.0.0

### PR DESCRIPTION
**Pull request by ShadowRZ**
###### Description of changes

https://www.renpy.org

Ren'Py is a visual novel engine written in Python.
This PR adds Ren'Py 8, which is based on Python 3.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -11461,6 +11461,12 @@
     githubId = 543055;
     name = "Shadaj Laddad";
   };
+  shadowrz = {
+    email = "23130178+ShadowRZ@users.noreply.github.com";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:11465]: Please use a none github email.

File: pkgs/development/interpreters/renpy/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    python module/setup.py install --prefix=$out --install-lib=$out/share/renpy/module
+
+    makeWrapper ${python}/bin/python $out/bin/renpy \
+      --set PYTHONPATH $PYTHONPATH:$out/share/renpy/module \
+      --add-flags "-O $out/share/renpy/renpy.py"
+  '';
+
+  NIX_CFLAGS_COMPILE = "-I${pygame_sdl2}/include/${python.libPrefix}";
+}
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/renpy/default.nix:60]: ~~meta is missing.~~ meta goes last, not after pname+version.

File: pkgs/development/interpreters/renpy/default.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  '';
+
+  installPhase = ''
+    mkdir -p $out/share/renpy
+    cp -vr * $out/share/renpy
+    rm -rf $out/share/renpy/module
+
+    python module/setup.py install --prefix=$out --install-lib=$out/share/renpy/module
+
+    makeWrapper ${python}/bin/python $out/bin/renpy \
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/renpy/default.nix:54]:
```suggestion
    makeWrapper ${python.interpreter} $out/bin/renpy \
```

File: pkgs/development/interpreters/renpy/default.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    SDL2 SDL2.dev libpng ffmpeg ffmpeg.out freetype glew.dev glew.out libGLU libGL fribidi zlib
+  ]);
+
+  patchPhase = ''
+    cp tutorial/game/tutorial_director.rpy{m,}
+  '';
+
+  buildPhase = ''
+    python module/setup.py build
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/renpy/default.nix:45]: This ignores $NIX_BUILD_CORES.

File: pkgs/development/interpreters/renpy/default.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    SDL2 libpng ffmpeg freetype glew libGLU libGL fribidi zlib pygame_sdl2 glib
+  ];
+
+  pythonPath = [ pygame_sdl2 tkinter future six pefile requests ];
+
+  RENPY_DEPS_INSTALL = lib.concatStringsSep "::" (map (path: path) [
+    SDL2 SDL2.dev libpng ffmpeg ffmpeg.out freetype glew.dev glew.out libGLU libGL fribidi zlib
+  ]);
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/renpy/default.nix:39]:
```suggestion
  postPatch = ''
```

File: pkgs/development/interpreters/renpy/default.nix [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  src = fetchFromGitHub {
+    owner = "renpy";
+    repo = "renpy";
+    rev = "8.0.0.22062402";
+    sha256 = "sha256-37Hbs0i5eXMjVaETX7ImJCak0y8XtEHUaRFceA9J39A=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [
+    python cython wrapPython tkinter future six pefile requests
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/renpy/default.nix:29]: cython and wrapPython goes to nativeBuildInputs.

File: pkgs/development/interpreters/renpy/default.nix [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    SDL2 libpng ffmpeg freetype glew libGLU libGL fribidi zlib pygame_sdl2 glib
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/renpy/default.nix:30]: Are those actually used at compile time or only through wrapping in RENPY_DEPS_INSTALL? If the second we can drop them from here.

File: pkgs/development/interpreters/renpy/default.nix [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    homepage = "https://renpy.org/";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ shadowrz ];
+  };
+
+  src = fetchFromGitHub {
+    owner = "renpy";
+    repo = "renpy";
+    rev = "8.0.0.22062402";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/renpy/default.nix:23]:
```suggestion
    rev = version;
```

File: pkgs/development/interpreters/renpy/default.nix [hunk 9]
```diff
@@ -0,0 +1,60 @@
+{ lib, stdenv, fetchFromGitHub, python3Packages, pkg-config, SDL2
+, libpng, ffmpeg, freetype, glew, libGL, libGLU, fribidi, zlib
+, glib
+}:
+
+with python3Packages;
+
+stdenv.mkDerivation rec {
+  pname = "renpy";
+  version = "8.0.0";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/renpy/default.nix:10]:
```suggestion
  version = "8.0.0.22062402";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/renpy/default.nix (hunk 7 above)
Comment (ShadowRZ) [reply pkgs/development/interpreters/renpy/default.nix:27]: They are used actually. I tried to drop them here, but the build system couldn't find the headers.

File: pkgs/development/interpreters/renpy/default.nix (hunk 4 above)
Comment (ShadowRZ) [reply pkgs/development/interpreters/renpy/default.nix]: Changed to use `setuptoolsBuildHook` which should respect `$NIX_BUILD_CORES`.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (ShadowRZ) [reply maintainers/maintainer-list.nix:11465]: There are other maintainers that uses a email like this. Actually the commit https://github.com/NixOS/nixpkgs/commit/64eb5ec86305e6547d5a7a37efda936584fbbafc uses such email address.

Also:
```console
$ sed -ne '/users.noreply.github.com/p' maintainers/maintainer-list.nix | wc -l
59
```

Anyway if this is undesired I'll change that.

Comment (ShadowRZ) [reply maintainers/maintainer-list.nix]: Changed.

**Review by 7c6f434c — APPROVED** [review]

File: pkgs/development/interpreters/renpy/default.nix [hunk 10]
```diff
@@ -0,0 +1,79 @@
+{ lib, stdenv, fetchFromGitHub, python3Packages, pkg-config, SDL2
+, libpng, ffmpeg, freetype, glew, libGL, libGLU, fribidi, zlib
+, glib
+}:
+
+with python3Packages;
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/renpy/default.nix:6]: Explicitly list the Python 3 packages you are using.

File: pkgs/development/interpreters/renpy/default.nix [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+    makeWrapper ${python.interpreter} $out/bin/renpy \
+      --set PYTHONPATH $PYTHONPATH \
+      --add-flags "-O $out/share/renpy/renpy.py"
+  '';
+
+  NIX_CFLAGS_COMPILE = "-I${pygame_sdl2}/include/${python.libPrefix}";
+
+  meta = with lib; {
+    description = "Ren'Py Visual Novel Engine";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/renpy/default.nix:73]:
```suggestion
    description = "Visual Novel Engine";
```

File: pkgs/development/python-modules/pygame_sdl2/default.nix [hunk 12]
```diff
@@ -4,12 +4,12 @@
...
   pname = "pygame_sdl2";
   version = "2.1.0";
-  renpy_version = "7.2.0";
+  renpy_version = "8.0.0";
   name = "${pname}-${version}-${renpy_version}";
 
   src = fetchurl {
     url = "https://www.renpy.org/dl/${renpy_version}/pygame_sdl2-${version}-for-renpy-${renpy_version}.tar.gz";
-    sha256 = "1amgsb6mm8ssf7vdcs5dr8rlxrgyhh29m4i573z1cw61ynd7vgcw";
+    sha256 = "0majf64pdfba5byjlv41pgsdmwvy09hw3m7143jz3kc1wjd2gaw8";
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/pygame_sdl2/default.nix:12]: Could you change this to SRI style hash?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pygame_sdl2/default.nix (hunk 12 above)
Comment (ShadowRZ) [reply pkgs/development/python-modules/pygame_sdl2/default.nix]: Sure.

File: pkgs/development/python-modules/pygame_sdl2/default.nix [hunk 13]
```diff
@@ -4,12 +4,12 @@
 buildPythonPackage rec {
   pname = "pygame_sdl2";
   version = "2.1.0";
-  renpy_version = "7.2.0";
+  renpy_version = "8.0.0";
   name = "${pname}-${version}-${renpy_version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pygame_sdl2/default.nix:8]:
```suggestion
```
Can we get this from renpy? Something like ``${lib.versions.majorMinor renpy.version}.${lib.versions.patch renpy.version}

File: pkgs/development/interpreters/renpy/default.nix [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+    cat > renpy/vc_version.py << EOF
+    vc_version = ${vc_version}
+    official = False
+    nightly = False
+    EOF
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    python module/setup.py build --parallel=$NIX_BUILD_CORES
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/renpy/default.nix:55]:
```suggestion
    ${python.interpreter} module/setup.py build --parallel=$NIX_BUILD_CORES
```

File: pkgs/development/python-modules/pygame_sdl2/default.nix (hunk 13 above)
Comment (ShadowRZ) [reply pkgs/development/python-modules/pygame_sdl2/default.nix]: Done.

**Pull request merged**